contracts/src/v0.8/vrf/dev/SubscriptionAPI.sol

```diff
@@ -396,36 +401,42 @@ abstract contract SubscriptionAPI is ConfirmedOwner, IERC677Receiver, IVRFSubscr
    */
   function addConsumer(uint256 subId, address consumer) external override onlySubOwner(subId) nonReentrant {
     // Already maxed, cannot add any more consumers.
-    if (s_subscriptionConfigs[subId].consumers.length == MAX_CONSUMERS) {
+    address[] storage consumers = s_subscriptionConfigs[subId].consumers;
+    if (consumers.length == MAX_CONSUMERS) {
       revert TooManyConsumers();
     }
-    if (s_consumers[consumer][subId] != 0) {
+    mapping(uint256 => uint64) storage nonces = s_consumers[consumer];
+    if (nonces[subId] != 0) {
       // Idempotence - do nothing if already added.
       // Ensures uniqueness in s_subscriptions[subId].consumers.
       return;
     }
     // Initialize the nonce to 1, indicating the consumer is allocated.
-    s_consumers[consumer][subId] = 1;
-    s_subscriptionConfigs[subId].consumers.push(consumer);
+    nonces[subId] = 1;
+    consumers.push(consumer);
 
     emit SubscriptionConsumerAdded(subId, consumer);
   }
 
   function _deleteSubscription(uint256 subId) internal returns (uint96 balance, uint96 nativeBalance) {
-    SubscriptionConfig memory subConfig = s_subscriptionConfigs[subId];
-    Subscription memory sub = s_subscriptions[subId];
-    balance = sub.balance;
-    nativeBalance = sub.nativeBalance;
+    address[] storage consumers = s_subscriptionConfigs[subId].consumers;
```

Comment:
just making sure, this is more gas efficient than `address[] memory consumers` right?

Reply:
Yes, I tried both `storage` and `memory`.

Using `storage`, `testMigration() (gas: -1221 (-0.256%))`
Using `memory`, `testMigration() (gas: -1086 (-0.228%))`

In this case, it's cheaper to access storage repeatedly than to copy out all the consumers into memory.